src/modules/sub-items/sub.items.module.js

```diff
@@ -60,9 +74,26 @@ export default class SubItemsModule extends Component {
         document.body.appendChild(this.bulkDeleteModalContainer);
     }
 
-    componentDidUpdate(prevProps, prevState) {
-        if (prevState.offset < this.state.offset) {
-            this.loadItems();
+    componentDidUpdate() {
+        const { activePage, pages, totalCount } = this.state;
+        const { limit: itemsPerPage } = this.props;
+        const numberOfPages = Math.ceil(totalCount / itemsPerPage);
+        const pageDoesNotExist = activePage > numberOfPages - 1 && activePage !== 0;
+
+        if (pageDoesNotExist) {
+            this.setState({
```

Comment:
This is risky, as it might lead to infinite loop. We have to be really careful when updating state in the lifecycle method that is run right after a state is updated.